notebooks/406-3D-pose-estimation-webcam/406-opencv-3D-pose-estimation.ipynb

```diff
@@ -0,0 +1,587 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "22fd3e08-ed3b-44ac-bd07-4a80130d6681",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # 3D edge index array\n",
+    "body_edges = np.array(\n",
+    "    [\n",
+    "        [11, 10],\n",
+    "        [10, 9],\n",
+    "        [9, 0],\n",
+    "        [0, 3],\n",
+    "        [3, 4],\n",
+    "        [4, 5],\n",
+    "        [0, 6],\n",
+    "        [6, 7],\n",
+    "        [7, 8],\n",
+    "        [0, 12],\n",
+    "        [12, 13],\n",
+    "        [13, 14],\n",
+    "        [0, 1],\n",
+    "        [1, 15],\n",
+    "        [15, 16],\n",
+    "        [1, 17],\n",
+    "        [17, 18],\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "body_edges_2d = np.array(\n",
+    "    [\n",
+    "        [0, 1],  # neck - nose\n",
+    "        [1, 16],\n",
+    "        [16, 18],  # nose - l_eye - l_ear\n",
+    "        [1, 15],\n",
+    "        [15, 17],  # nose - r_eye - r_ear\n",
+    "        [0, 3],\n",
+    "        [3, 4],\n",
+    "        [4, 5],  # neck - l_shoulder - l_elbow - l_wrist\n",
+    "        [0, 9],\n",
+    "        [9, 10],\n",
+    "        [10, 11],  # neck - r_shoulder - r_elbow - r_wrist\n",
+    "        [0, 6],\n",
+    "        [6, 7],\n",
+    "        [7, 8],  # neck - l_hip - l_knee - l_ankle\n",
+    "        [0, 12],\n",
+    "        [12, 13],\n",
+    "        [13, 14],\n",
+    "    ]\n",
+    ")  # neck - r_hip - r_knee - r_ankle\n",
+    "\n",
+    "\n",
+    "def draw_poses(img, poses_2d):\n",
+    "    for pose in poses_2d:\n",
+    "        pose = np.array(pose[0:-1]).reshape((-1, 3)).transpose()\n",
+    "\n",
+    "        was_found = pose[2] > 0\n",
+    "\n",
+    "        # Draw joints.\n",
+    "        for edge in body_edges_2d:\n",
+    "            if was_found[edge[0]] and was_found[edge[1]]:\n",
+    "                cv2.line(\n",
+    "                    img,\n",
+    "                    tuple(pose[0:2, edge[0]].astype(np.int32)),\n",
+    "                    tuple(pose[0:2, edge[1]].astype(np.int32)),\n",
+    "                    (255, 255, 0),\n",
+    "                    4,\n",
+    "                    cv2.LINE_AA,\n",
+    "                )\n",
+    "        # Draw limbs.\n",
+    "        for kpt_id in range(pose.shape[1]):\n",
+    "            if pose[2, kpt_id] != -1:\n",
+    "                cv2.circle(\n",
+    "                    img,\n",
+    "                    tuple(pose[0:2, kpt_id].astype(np.int32)),\n",
+    "                    3,\n",
+    "                    (0, 255, 255),\n",
+    "                    -1,\n",
+    "                    cv2.LINE_AA,\n",
+    "                )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c79ffb0f-f386-4e7b-8cbf-2ba2fe5607d0",
+   "metadata": {},
+   "source": [
+    "### 3D Visualization library\n",
+    "A 3D visualization library is introduced to map 3D coordinates onto a 2D screen.\n",
```

Comment:
Whats the connection between this words and the code block below for normalization ? Could help to add some notes to explain this isolated code block as well？